### PR TITLE
[JENKINS-44851] Agents JVM must be 1.8+ now

### PR DESCRIPTION
**Pull request by amuniz**
# Description

See [JENKINS-44851](https://issues.jenkins-ci.org/browse/JENKINS-44851).

Not sure how to test this, maybe an ATH with `@DockerFixture`, but it would be a separate PR.

### Changelog entries

Proposed changelog entries:

* Agents JVM must be 1.8+ and a clear message is shown in connection logs if it is not.

### Submitter checklist

- [x] JIRA issue is well described
- [x] Link to JIRA ticket in description, if appropriate
- [x] Appropriate autotests or explanation to why this change has no tests
- [ ] For new API and extension points: Link to the reference implementation in open-source (or example in Javadoc)

### Desired reviewers

@reviewbybees

### CODE REVIEW

**Review by jglick — APPROVED** [review]
With the caveat mentioned in JIRA that this will not currently affect `ssh-slaves`.

**Comment by daniel-beck** [comment]
Is there no way to plug the `target.version` from the POM in there? Like we do with the changelog URL via filtered resources? Otherwise we'll be back here in two years.

**Review by oleg-nenashev — APPROVED** [review]
🐝

**Comment by daniel-beck** [comment]
CC @olivergondza this may qualify as a late inclusion in 2.60.1.

**Comment by oleg-nenashev** [comment]
Ideally Jenkins core should expose a public constant like `Jenkins.REQUIRED_JAVA_VERSION`, which can be used in both core and plugins. Otherwise we may have the same fallout once we move to Java 9... seriously.

I am trying to track all issues we had in https://issues.jenkins-ci.org/browse/JENKINS-43500

**Comment by olivergondza** [comment]
I agree with Daniel this would better be inferred rather than hardcoded. Also, since it was checking for `1.6` until this point, I do not thing there is a hurry to fix it this late in LTS cycle.

**Comment by oleg-nenashev** [comment]
Should be merged along with https://github.com/jenkinsci/jenkins/pull/2917, which adds similar check to the Executable war

**Comment by daniel-beck** [comment]
@olivergondza TBH I would prefer this be merged and backported now, and then we think about how to future-proof it.

**Review by olivergondza — APPROVED** [review]
Ok, Let's merge it as it is.

**Comment by daniel-beck** [comment]
Retriggered the PR build.

**Review by oleg-nenashev — CHANGES_REQUESTED** [review]
:bug: Test failure is valid. Tests need to be fixed

**Comment by oleg-nenashev** [comment]
Tests screenshot:

screen shot 2017-06-16 at 15 25 47

**Comment by oleg-nenashev** [comment]
@amuniz  Just in case, do you plan to finalize it soon? The fix missed 2.60.1, but there is still chance for 2.60.2

**Comment by amuniz** [comment]
Mmm, I wonder how it worked before. Will have a look as soon as I can.

**Comment by daniel-beck** [comment]
@amuniz Ping

**Comment by amuniz** [comment]
@oleg-nenashev @daniel-beck pong

**Review by jglick — APPROVED** [review]

**Review by daniel-beck — APPROVED** [review]

**Review by johnou — APPROVED** [review]

File: core/src/test/java/hudson/slaves/ComputerLauncherTest.java [hunk 1]
```diff
@@ -36,15 +36,15 @@
 
 public class ComputerLauncherTest {
 
-    @Test public void jdk7() throws IOException {
+    @Test(expected=IOException.class) public void jdk7() throws IOException {
```
Comment (oleg-nenashev) [inline core/src/test/java/hudson/slaves/ComputerLauncherTest.java]: Exception on a separate line would be preferable

**Review by oleg-nenashev — APPROVED** [review]
🐝 and @reviewbybees done. Too late for this weekly, right?

File: core/src/test/java/hudson/slaves/ComputerLauncherTest.java (hunk 1 above)
Comment (amuniz) [reply core/src/test/java/hudson/slaves/ComputerLauncherTest.java]: Just followed existing style in this file.

**Comment by oleg-nenashev** [comment]
:ship::it:

**Pull request merged**